src/utils/index.tsx

```diff
@@ -0,0 +1,41 @@
+export const parseHeader = (header: any) => {
+  const macaroonRegex = /macaroon="([^"]+)"/;
+  const invoiceRegex = /invoice="([^"]+)"/;
+
+  const macaroonMatch = header?.match(macaroonRegex);
+  const invoiceMatch = header?.match(invoiceRegex);
+
+  return {
+    macaroon: macaroonMatch ? macaroonMatch[1] : null,
+    invoice: invoiceMatch ? invoiceMatch[1] : null,
+  };
+};
+
+export const isValidUrl = (url: string): boolean => {
+  const pattern = /^(ftp|http|https):\/\/[^ "]+$/;
```

Comment:
would have been good to allow for URLs without https:// and add it automatically, but this has complications of its own so this approach is acceptable for an initial MVP.